scipy/stats/_continuous_distns.py

```diff
@@ -8426,6 +8433,17 @@ def _stats(self, beta):
     def _entropy(self, beta):
         return 1. / beta - np.log(.5 * beta) + sc.gammaln(1. / beta)
 
+    def _rvs(self, beta, size=None, random_state=None):
+        # see [2]_ for the algorithm
+        # see [3]_ for reference implementation in SAS
+        z = random_state.gamma(1/beta, size=size)
+        y = z ** (1/beta)
+        # convert y to array to ensure masking support
+        y = np.asarray(y)
+        mask = random_state.random(size=y.shape) < 0.5
+        y[mask] = -y[mask]
+        return y
```

Comment:
This looks very much like the procedure in [2].
![image](https://user-images.githubusercontent.com/6570539/149585331-5327b517-c501-4d1a-93e3-c65e680ddc90.png)

i) is `z = random_state.gamma(1/beta, size=size)` (but I have the question about our one-parameter gamma distribution and _A_ above)
ii) is `y = z ** (1/beta)`
iii) is implemented by `mask = random_state.random(size=y.shape) < 0.5; y[mask] = -y[mask]`. Perhaps the comments after equation 9 make it more clear:
![image](https://user-images.githubusercontent.com/6570539/149585655-412f9f4e-205b-481f-bb0f-1668c54b04eb.png)

Reply:
This is all correct. I answered the question about the one-parameter gamma distribution and A [here](https://github.com/scipy/scipy/pull/15413#discussion_r785278022).